modules/extensions/src/terrain/terrain-cover.ts

```diff
@@ -0,0 +1,240 @@
+import {Framebuffer} from '@luma.gl/core';
+
+import {WebMercatorViewport, OrthographicViewport} from '@deck.gl/core';
+import type {Layer, Viewport} from '@deck.gl/core';
+
+import {createRenderTarget} from './utils';
+
+const unitGeospatialViewport = new WebMercatorViewport({width: 1, height: 1});
+const unitNonGeospatialViewport = new OrthographicViewport({width: 1, height: 1});
+
+const MAX_SIZE = 2048;
+
+// TODO - import from loaders when Tileset2D is split out
```

Comment:
FYI @ibgreen - I'm not importing the types from `geo-layers` because `geo-layers` currently depend on `extensions`. TypeScript really does not like circular dependencies. Also a good argument to strip `Tileset2D` out of that increasingly complex module.

Reply:
As long as you use `import type` you might be OK, as TypeScript does support for circular `import type` declarations.

But I agree we should try very hard to avoid this circularity as it is really confusing and will lead to other things creeping in down the line. 

BTW, I got a good way on the PR separating Tileset2D from Viewport but then ran into issues:

- both node and browser tests are badly broken on Apple M1, so I really couldn't iterate in an efficient way, replying on CI takes like 10-15 minutes per cycle.
- Also there is one part where the Tileset2D interacts back and forth with the viewport which makes it hard to make the required values static input parameters, would be great to get your thoughts on that.